docs/src/modules/components/ApiPage/sections/ToggleDisplayOption.tsx

```diff
@@ -42,10 +44,39 @@ const getRandomOption = () => {
   return randomOption;
 };
 
+let neverHydrated = true;
+
+const getOption = (storageKey: string) => () => {
+  if (neverHydrated) {
+    return DEFAULT_LAYOUT;
+  }
+  try {
+    const savedOption = localStorage.getItem(storageKey);
+
+    if (savedOption !== null) {
+      return savedOption as ApiDisplayOptions;
+    }
+
+    const randomOption = getRandomOption();
+
+    return randomOption;
+  } catch (error) {
+    return DEFAULT_LAYOUT;
+  }
+};
+
 export function useApiPageOption(
   storageKey: string,
 ): [ApiDisplayOptions, (newOption: ApiDisplayOptions) => void] {
-  const [option, setOption] = React.useState(options[0]);
+  const [option, setOption] = React.useState(getOption(storageKey));
+  useEnhancedEffect(() => {
+    if (neverHydrated) {
+      if (document?.location.hash) {
+        document.getElementById(document?.location.hash)?.scrollIntoView();
+      }
```

Comment:
The location should be defined no?

```suggestion
      document.getElementById(document.location.hash)?.scrollIntoView();
```

or if it's really needed:

```suggestion
      document.getElementById(document!.location.hash)?.scrollIntoView();
```